packages/php-wasm/web-service-worker/src/initialize-service-worker.ts

```diff
@@ -22,28 +32,28 @@ export function initializeServiceWorker(config: ServiceWorkerConfiguration) {
 
 		// Don't handle requests to the service worker script itself.
 		if (url.pathname.startsWith(self.location.pathname)) {
-			return;
+			// return;
 		}
 
 		// Only handle requests from scoped sites.
 		// So – bale out if the request URL is not scoped and the
 		// referrer URL is not scoped.
-		if (!isURLScoped(url)) {
-			let referrerUrl;
-			try {
-				referrerUrl = new URL(event.request.referrer);
-			} catch (e) {
-				return;
-			}
-			if (!isURLScoped(referrerUrl)) {
-				// Let the browser handle uncoped requests as is.
-				return;
-			}
+		if (
+			!isURLScoped(url) &&
+			!(
+				event.request.referrer &&
+				isURLScoped(new URL(event.request.referrer))
```

Comment:
This referrer may not be a valid URL and we still need that try/catch:

> The referrer read-only property of the [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) interface is set by the user agent to be the referrer of the Request. (e.g., `client`, `no-referrer`, or a URL.)

Reply:
Thanks! I didn't know that [c34fd37](https://github.com/WordPress/wordpress-playground/pull/1535/commits/c34fd37ca20b4990899915df479c8e6357fdb815)